File: 20-valid-parentheses/valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []

        for i in s:
            if i in ['(','[','{']:
                stack.append(i)
            elif i == ')' and len(stack)!=0 and stack[-1] == '(':
                stack.pop()
            elif i == '}' and len(stack)!=0 and stack[-1] == '{':
                stack.pop()
            elif i == ']' and len(stack)!=0 and stack[-1] == '[':
                stack.pop()
            else:
                return False
        
        if len(stack)==0 :
            return True
        return False

File: 20-valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String str) {
        Stack<Character> stack = new Stack<>();
        char ch;
        char[] arr = str.toCharArray();

        for(int i = 0; i<str.length();i++){
            ch = arr[i];
            if(ch=='(' || ch == '[' || ch =='{'){
                stack.push(ch);
                continue;
            }

            if(stack.empty()) return false;
            if(ch == ')' && stack.pop() != '(') return false;
            if(ch == '}' && stack.pop() != '{') return false;
            if(ch == ']' && stack.pop() != '[') return false;

        }
        if(!stack.empty()) return false;
        return true;
    }
}